File: backend/spotify-server/Models/Playlist/Playlist.cs
namespace spotify_server.Models.Playlist;

public class Playlist : IDefaultItem
{
     public string type {get;set;}
    public long id {get;set;}
    public long[] owner {get;set;}
    public string title {get;set;}
    public string image {get;set;}
    public long[] contents {get;set;}
    public DateTime date {get;set;}
}

File: frontend/src/models/Items/Playlist.ts
import ID from "../ID";
import ItemType from "../ItemType";
import Track, { TrackID } from "./Track";
import { ArtistID, ArtistView } from "./Artist";
import {ImageURL} from "../url";
import DefaultItem from "../DefaultItem";

export type PlaylistContent = Track;

interface Playlist extends DefaultItem {
    type: ItemType.PLAYLIST;
    id: PlaylistID;
    title: string;
    image?: ImageURL;
    owner: ArtistView[];
    contents: PlaylistContent[]
}

export type PlaylistID = ID;

export default Playlist